### PR TITLE
doc: Add Azure Devops doc

### LINKED ISSUE #500: [Doc][Azure] Document Azure DevOps
## Documentation Scope

- [ ] add a plugin doc here with the name 'Azure DevOps(Beta)': https://devlake.apache.org/docs/Plugins.
- [ ] add a manual doc here with the name 'Azure DevOps(Beta)': https://devlake.apache.org/docs/Configuration/Tutorial

Please follow the doc structure for other plugin docs and manuals.

### PR DESCRIPTION
**Pull request by CamilleTeruel**
# Summary

Add documentation for Azure Devops plugin.

### Does this close any open issues?
Closes #500 

@Startrekzky

### CODE REVIEW

**Review by hezyin — APPROVED** [review]
LGTM

File: docs/Overview/SupportedDataSources.md [hunk 1]
```diff
@@ -9,81 +9,82 @@ sidebar_position: 5
...
+| AE               | Source Code Management                                                       | On-prem                            | Not Available           | `ae`                           | Full Refresh                                                    |
+| Sonarqube        | CODE QUALITY                                                                 | SonarQube v8.x, v9.x               | Available               | `sonarqube`                    | Full Refresh                                                    |
+| Bamboo CI(WIP)   | CI/CD                                                                        | v6.8.1                             | Not Available           | `bamboo`                       | Full Refresh                                                    |
+| Bamboo CI(WIP)   | CI/CD, Source Code Management, Code Review                                   | Cloud                              | Available               | `azuredevops`, `gitextractor`  | Full Refresh                                                    |
```
Comment (hezyin) [inline docs/Overview/SupportedDataSources.md]: @CamilleTeruel Seems like you meant "Azure DevOps (WIP)" instead of "Bamboo CI (WIP)" here.

**Review by hezyin — CHANGES_REQUESTED** [review]
Only one minor comment

**Review by hezyin — APPROVED** [review]
LGTM

**Comment by aFlyBird0** [comment]
@CamilleTeruel 
Hello, the CI error (Link Checker) has most likely been fixed in #510. Could you please try `git switch 500_AzureDevOps_doc && git fetch upstream && git rebase upstream/main && git push` to pull the latest code and re-trigger CI? Thank you!

**Pull request merged**